statistics/handle/cache/internal/lfu/lfu_cache.go

```diff
@@ -78,11 +78,15 @@ func (s *LFU) Get(tid int64, _ bool) (*statistics.Table, bool) {
 
 // Put implements statsCacheInner
 func (s *LFU) Put(tblID int64, tbl *statistics.Table) bool {
-	ok := s.cache.Set(tblID, tbl, tbl.MemoryUsage().TotalTrackingMemUsage())
-	if ok { // NOTE: `s.cache` and `s.resultKeySet` may be inconsistent since the update operation is not atomic, but it's acceptable for our scenario
-		s.resultKeySet.AddKeyValue(tblID, tbl)
-		s.cost.Add(tbl.MemoryUsage().TotalTrackingMemUsage())
-	}
+	cost := tbl.MemoryUsage().TotalTrackingMemUsage()
+	// Here we need to insert resultKeySet first and then write to LFU,
+	// in order to prevent data race. If the LFU cost is already full,
```

Comment:
We have lock in `AddKeyValue`, so looks like there should be no data races?

Reply:
In fact, I found the data race in the cache item. so we must not happen the operators on the cache item at the same time.

Reply:
If we ```AddKeyValue``` later, we do the reject and ```AddKeyValue``` simultaneously.